src/workflows/recipes/pages/RecipeListingPage.js

```diff
@@ -178,12 +105,100 @@ class RecipeListingPage extends React.PureComponent {
   }
 
   render() {
-    const { columns, count, ordering, pageNumber, recipes, status } = this.props;
+    const { columns, count, ordering, pageNumber, recipes, status, action, actions } = this.props;
+    const actionNames = Object.values(actions.toJS()).map(action => action.name);
+    actionNames.sort();
 
     const filters = this.getFilters();
 
     const filterIds = Object.keys(filters).map(key => `${key}-${filters[key]}`);
     const requestId = `fetch-filtered-recipes-page-${pageNumber}-${filterIds.join('-')}`;
+
```

Comment:
The change here is that I moved it from being a static object of functions to instead being defined *inside* the `render` method. I did that so I can dynamically (using closures) get the list of all possible action names. And I also, changed the function for `action`. The rest is just code moves.